pre-commit: add pre-commit

m-iwanicki opened the pull request:
Adds pre-commit without fixes.
We could also add link checker as in here? https://github.com/TrenchBoot/TrenchBoot.github.io/blob/d52e8f54b924d94cdc97437117a1c3e2095fead9/.pre-commit-config.yaml#L58

Probably README.md needs to be modified

DaniilKl commented on .codespellrc:
```diff
@@ -0,0 +1,3 @@
+[codespell]
+exclude-file = .codespellx
+ignore-words-list = "FPT,FTP,fpt,ftp"
```
```suggestion
ignore-words-list = "FPT,FTP,fpt,ftp,SoM"
```

```bash
codespell................................................................Failed
- hook id: codespell
- exit code: 65

docs/supported-targets/radxa-cm3.md:179: SoM ==> some
```

DaniilKl replied:
@m-iwanicki, you should adhere to our internal review process. Actually we could publish it as a part of Zarhus docs.

DaniilKl replied:
I see it has been fixed here 9e62b6d06b3d743c554288354cec768d748d96af.

m-iwanicki replied:
![image](https://github.com/user-attachments/assets/2df96f08-1760-4e3c-b605-899b7bec3613)
This button automatically commits your suggestion and closes issue

DaniilKl commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,54 @@
+          - commit-msg
+
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.3.0
```
We have rev. `v2.2.5` in our `3mdeb/hooks` repo. Could you update it as well?

DaniilKl commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,54 @@
+        args: [--fix]
+
+  - repo: https://github.com/igorshubovych/markdownlint-cli
+    rev: v0.41.0
```
We have rev. `v0.36.0` in our `3mdeb/hooks` repo. Could you update it as well?

m-iwanicki replied:
`pre-commit-init.py` from 3mdeb/hooks automatically updates hooks to newest version during init.

DaniilKl reviewed (changes requested):
I see a lot of failed hooks, could you please fix them as well? Perhaps, while fixing, you will find out that we need some changes in hooks configs for this repo.

DaniilKl commented:
> We could also add link checker as in here? https://github.com/TrenchBoot/TrenchBoot.github.io/blob/d52e8f54b924d94cdc97437117a1c3e2095fead9/.pre-commit-config.yaml#L58
>
> Probably README.md needs to be modified

Yep and yep.

m-iwanicki commented:
```
~/projects/docs/zarhus-docs git:(add-pre-commit) ✗ pre-commit run --all-files
check for added large files..............................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
detect private key.......................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
mixed line ending........................................................Passed
codespell................................................................Passed
yamllint.................................................................Passed
markdownlint.............................................................Passed
markdownlint-fix.........................................................Passed
```

m-iwanicki commented:
lychee, should be fixed by someone who wrote those documents

```
lychee...................................................................Failed
- hook id: lychee-docker
- exit code: 2

✗ [404] https://fosstodon.org/@TBD | Failed: Network error: Not Found
✗ [404] https://cloud.3mdeb.com/index.php/s/irabKp2xKjPLH68/download;downloadfilename=GoDEXdriver-1.1.4.tar.gz | Failed: Network error: Not Found
✗ [404] https://files.pythonhosted.org/packages/8c/ca/e52321c8d9b4b01db6dd73d9df65ed011bd2199e780367faf2f6a8bf0229/ipaddresses-$%7BPV%7D.zip | Failed: Network error: Not Found
Issues found in 2 inputs. Find details below.

[mkdocs.yml]:
✗ [404] https://fosstodon.org/@TBD | Failed: Network error: Not Found

[docs/guides/managing-packages.md]:
✗ [404] https://files.pythonhosted.org/packages/8c/ca/e52321c8d9b4b01db6dd73d9df65ed011bd2199e780367faf2f6a8bf0229/ipaddresses-$%7BPV%7D.zip | Failed: Network error: Not Found
✗ [404] https://cloud.3mdeb.com/index.php/s/irabKp2xKjPLH68/download;downloadfilename=GoDEXdriver-1.1.4.tar.gz | Failed: Network error: Not Found

🔍 134 Total (in 1s) ✅ 45 OK 🚫 3 Errors 💤 86 Excluded
```

m-iwanicki commented:
@DaniilKl it might be happening because lychee checks comments (mkdocs.yml) and I also checked links in code blocks.
I added links in comments to `.lycheeignore` file and removed code block checking (I guess there would be too many false-positives)

m-iwanicki commented:
@DaniilKl Also uncommented pre-commit section in README.md.
Not sure if `- [Install go](https://go.dev/doc/install)` is needed but I left it as is.

DaniilKl commented:
> Not sure if - [Install go](https://go.dev/doc/install) is needed but I left it as is.

I do not have GO installed and `pre-commit` and all its hooks work well in this repo:

```bash
λ go version
zsh: go: command not found...
λ pre-commit run --all-files
check for added large files..............................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
detect private key.......................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
mixed line ending........................................................Passed
codespell................................................................Passed
yamllint.................................................................Passed
markdownlint.............................................................Passed
markdownlint-fix.........................................................Passed
lychee...................................................................Passed
```

So I will delete it from README then.

DaniilKl commented:
@m-iwanicki, thank you for fixing all issues here.
